kratos/geometries/line_2d_2.h

```diff
@@ -925,20 +925,44 @@ class Line2D2 : public Geometry<TPointType>
      */
     bool HasIntersection(const Point& rLowPoint, const Point& rHighPoint) override
     {
-        bool has_intersection = false;
+        const double tolerance = std::numeric_limits<double>::epsilon();
         // We get the local points
         const TPointType& first_point  = BaseType::GetPoint(0);
         const TPointType& second_point = BaseType::GetPoint(1);
-        // Check if point one is inside the bounding box
-        has_intersection = has_intersection || (
-                                    ( (first_point[0] >= rLowPoint[0] && first_point[0] <= rHighPoint[0])
-                                        && (first_point[1] >= rLowPoint[1] && first_point[1] <= rHighPoint[1]) ) // IF the first point is inside the box
-                                    ||
-                                    ( (second_point[0] >= rLowPoint[0] && second_point[0] <= rHighPoint[0])
-                                        && (second_point[1] >= rLowPoint[1] && second_point[1] <= rHighPoint[1]) ) // IF the second point is inside the box
-                                );
-
-        return has_intersection;
+
+        if (    // If one of the point is inside the box then there is an intersection. If none is inside then we check further.
+                ( (first_point[0] >= rLowPoint[0] && first_point[0] <= rHighPoint[0])
+                    && (first_point[1] >= rLowPoint[1] && first_point[1] <= rHighPoint[1]) ) // IF the first point is inside the box
+                ||
+                  ( (second_point[0] >= rLowPoint[0] && second_point[0] <= rHighPoint[0])
+                    && (second_point[1] >= rLowPoint[1] && second_point[1] <= rHighPoint[1]) ) // IF the second point is inside the box
+            )
+            return true;
+
+        double high_x = rHighPoint[0]; double high_y = rHighPoint[1];
+        double low_x = rLowPoint[0];   double low_y = rLowPoint[1];
+
+        double slope = (second_point[1] - first_point[1]) / ( second_point[0] - first_point[0] );
+
+        // Intersection with left vertical line of the box that is x = low_x
+        double y_1 = slope*( low_x - first_point[0] ) + first_point[1];
+        if(y_1 >= low_y - tolerance && y_1 <= high_y+tolerance) // If y intersection is between two y bounds there is an intersection
+            return true;
+        // Intersection with right vertical line of the box that is x = high_x
+        double y_2 = slope*( high_x - first_point[0] ) + first_point[1];
+        if(y_2 >= low_y - tolerance && y_2 <= high_y+tolerance) // If y intersection is between two y bounds there is an intersection
```

Comment:
```suggestion
        const double y_2 = slope*( high_x - first_point[0] ) + first_point[1];
```

Reply:
why is this if condition deleted ?

Reply:
Huum, probably i did the suggestion wrong, I just wanted to add const